pkg/emptydisk/emptydisk.go

```diff
@@ -13,33 +13,37 @@ import (
 var EmptyDiskBaseDir = "/var/run/libvirt/empty-disks/"
 
 func CreateTemporaryDisks(vmi *v1.VirtualMachineInstance) error {
-
 	for _, volume := range vmi.Spec.Volumes {
-
 		if volume.EmptyDisk != nil {
-			// qemu-img takes the size in bytes or in Kibibytes/Mebibytes/...; lets take bytes
-			size := strconv.FormatInt(volume.EmptyDisk.Capacity.ToDec().ScaledValue(0), 10)
-			file := FilePathForVolumeName(volume.Name)
-			if err := os.MkdirAll(EmptyDiskBaseDir, 0777); err != nil {
-				return err
-			}
-			if _, err := os.Stat(file); os.IsNotExist(err) {
-				// #nosec No risk for attacket injection. Parameters are predefined strings
-				if err := exec.Command("qemu-img", "create", "-f", "qcow2", file, size).Run(); err != nil {
-					return err
-				}
-			} else if err != nil {
-				return err
-			}
-			if err := ephemeraldiskutils.DefaultOwnershipManager.SetFileOwnership(file); err != nil {
+			if err := createEmptyDiskForVolume(volume); err != nil {
 				return err
 			}
 		}
 	}
+	return nil
+}
 
+func createEmptyDiskForVolume(volume v1.Volume) error {
+	// qemu-img takes the size in bytes or in Kibibytes/Mebibytes/...; lets take bytes
+	size := strconv.FormatInt(volume.EmptyDisk.Capacity.ToDec().ScaledValue(0), 10)
+	file := FilePathForVolumeName(volume.Name)
```

Comment:
Could you just move the code a bit so that each declaration is located immediately before being used ? 

That would make:
  - create dir
  - declare 'file'
  - if not fileExists(file)
  - declare size
  - create disk w/ size

I think it becomes slightly more readable.

Reply:
You're right, this reads a bit better. [Updated](https://github.com/kubevirt/kubevirt/compare/3cf1af6c24d5ab956dd90a8256171c6487fd9e1b..bdaf43f298c5c940946dc5d4f7dd5b55ee823b22)